File: app/src/main/java/com/sevtinge/cemiuiler/module/mishare/MiShareDexKit.java
package com.sevtinge.cemiuiler.module.mishare;

import com.sevtinge.cemiuiler.module.base.BaseHook;
import io.luckypray.dexkit.DexKitBridge;
import io.luckypray.dexkit.builder.BatchFindArgs;
import io.luckypray.dexkit.descriptor.member.DexMethodDescriptor;
import io.luckypray.dexkit.enums.MatchType;

import java.util.List;
import java.util.Map;

public class MiShareDexKit extends BaseHook {

    public static Map<String, List<DexMethodDescriptor>> mMiShareResultMethodsMap;

    @Override
    public void init() {
        System.loadLibrary("dexkit");
        String apkPath = lpparam.appInfo.sourceDir;
        DexKitBridge bridge = DexKitBridge.create(apkPath);
        try {
            if (bridge == null) {
                return;
            }
            mMiShareResultMethodsMap =
                    bridge.batchFindMethodsUsingStrings(
                            BatchFindArgs.builder()
                                    .addQuery("MiShareAutoOff", List.of("MiShareService", "EnabledState"))
                                    .addQuery("qwq", List.of("EnabledState", "mishare_enabled"))
                                    .addQuery("qwq2", List.of("null context", "cta_agree"))
                                    .matchType(MatchType.CONTAINS)
                                    .build()
                    );
            } catch (Throwable e) {
            e.printStackTrace();
        }
        bridge.close();
    }
}

File: app/src/main/java/com/sevtinge/cemiuiler/module/mishare/MiShareDexKit.kt
package com.sevtinge.cemiuiler.module.mishare

import com.sevtinge.cemiuiler.module.base.BaseHook
import com.sevtinge.cemiuiler.utils.DexKit.closeDexKit
import com.sevtinge.cemiuiler.utils.DexKit.initDexKit
import com.sevtinge.cemiuiler.utils.DexKit.safeDexKitBridge
import io.luckypray.dexkit.descriptor.member.DexClassDescriptor
import io.luckypray.dexkit.descriptor.member.DexMethodDescriptor
import io.luckypray.dexkit.enums.MatchType

class MiShareDexKit : BaseHook() {
    override fun init() {
        System.loadLibrary("dexkit")
        initDexKit(lpparam)
        try {
            mMiShareResultMethodsMap = safeDexKitBridge.batchFindMethodsUsingStrings {
                addQuery("qwq", setOf("EnabledState", "mishare_enabled"))
                matchType = MatchType.FULL
            }
            mMiShareResultClassMap = safeDexKitBridge.batchFindClassesUsingStrings {
                addQuery("qwq2", setOf("null context", "cta_agree"))
                matchType = MatchType.FULL
            }
        } catch (e: Throwable) {
            e.printStackTrace()
        }
        closeDexKit()
    }

    companion object {
        var mMiShareResultMethodsMap: Map<String, List<DexMethodDescriptor>>? = null
        var mMiShareResultClassMap: Map<String, List<DexClassDescriptor>>? = null
    }
}
